tests/StockTest.php

```diff
@@ -73,4 +71,30 @@ public function testAddMultipleElements()
         $this->assertEquals($stock_array[1], $element2);
         $this->assertEquals(2, count($stock_array));
     }
+
+    public function testSuccessfulElementValidation()
+    {
+        $element = ['name' => 'brakes', 'price' => 199, 'producent' => 'acme'];
+
+        $this->stock->add($element);
+        $class = new ReflectionClass($this->stock);
+
+        $method = $class->getMethod('validate');
+        $method->setAccessible(true);
+        $is_valid = $method->invokeArgs($this->stock, [$element]);
+        $this->assertTrue($is_valid);
+    }
+
+    public function testFailureElementValidation()
+    {
+        $element = ['name' => 'brakes', 'price', 'producent' => 'acme'];
```

Comment:
Ta tablica jest myląca na pierwszy rzut oka. Nie wprowadzaj takiej niejasności.
Zamiast "nie ustawiać" wartości dla "price" to po prostu skasuj "price".
Będzie od razu wiadomo, że próbujesz wrzucić tylko dwa elementy, a nie trzy z czego jeden bez wartości.